orchagent/pfcactionhandler.cpp

```diff
@@ -266,10 +282,23 @@ PfcWdAclHandler::~PfcWdAclHandler(void)
 {
     SWSS_LOG_ENTER();
 
-    auto found = m_aclTables.find(m_strIngressTable);
-    found->second.unbind(getPort());
+    sai_object_id_t table_oid = gAclOrch->getTableById(m_strIngressTable);
+    map<sai_object_id_t, AclTable> table_map = gAclOrch->getAclTables();
+    auto rule_iter = table_map[table_oid].rules.find(m_strRule);
 
-    found = m_aclTables.find(m_strEgressTable);
+    vector<sai_object_id_t> port_set = rule_iter->second->getInPorts();
+    sai_object_id_t port = getPort();
+
+    if((port_set.size() == 1) && (port_set[0] == port))
+    {
+        gAclOrch->removeAclRule(m_strIngressTable, m_strRule);
+    }
+    else 
+    {
+        gAclOrch->updateAclRule(rule_iter->second, m_strIngressTable, MATCH_IN_PORTS, &port, RULE_OPER_DELETE);
+    } 
+
+    auto found = m_aclTables.find(m_strEgressTable);
     found->second.unbind(getPort());
```

Comment:
if Danny's comment on L247 holds good, then I think we need some extra handling for ingress table. what if the same port is bound to another rule for a diff queue?

Reply:
Sure. Will test out the L247 changes and make necessary changes here. Will probably unbind the port once there are no rules left in ingress table else we may nee do do walk of the table every time there is rule update.